src/components/post/index.js

```diff
@@ -62,6 +62,79 @@ const Post = ({
     }
   };
 
+  let likeIcon = null;
+  if (isLiked) {
+    likeIcon = <LikeRed />;
+  } else {
+    likeIcon = <Like />;
+  }
+
+    const showCommentModal = () => {
+        setModal('Add a comment...', 
+        <AddCommentModal 
+            setTriggerUpdate = {setTriggerUpdate}
+            id={id}
+            currentUserName={currentUserName}
+            currentUserInitials={currentUserInitials}
+            setUpdateComments={setUpdateComments}
+        />)
+        openModal()
+        }
+
+    const findComments = async () => {
+          const tempArray = [];
+          const res = await get(`posts/${id}/comments`);
+          const tempComments = res.data.comments;
+          await Promise.all(
```

Comment:
Why are you using Promise?

Reply:
Promise.all was used to ensure that all comments for a post have been received before it updates the state. Before using this method it was causing problems with the page attempting to render the comments before they were all available to render, so Promise.all was added to ensure they were all ready to be rendered.
Spoke to Lewis about how I implemented this yesterday afternoon and he thought it was an OK way to do it.

Reply:
Can you replicate this for me? It doesn't make sense why this would happen: once you remove the async callback to map() it's all synchronous.

Reply:
Ah wait, I remember!

Before Lewis updated the API structure the user details for the person who made the comment weren't included in the comment get, so for each comment I was having to run a get request to fetch the name of the user who made the comment, and that was what caused the issue and resulted in me changing it to a promise.all

Now that he's updated the API structure and I no longer need the additional get request for the user details I've removed the promise.all